GitCommands/Gpg/IGitGpgController.cs

```diff
@@ -0,0 +1,27 @@
+﻿using GitCommands.Gpg;
+
+namespace GitCommands.Git.Gpg
+{
+    public partial interface IGitGpgController
+    {
+        /// <summary>
+        /// Gets whether commits are signed by default
+        /// </summary>
+        /// <returns>true for default to signed commits</returns>
+        bool GetCommitGPGSign();
+
+        /// <summary>
+        /// Gets the user's default signing key from git config
+        /// </summary>
+        /// <returns>The key id of the gpg key to sign with.</returns>
+        string GetDefaultKey();
+
+        Task<IEnumerable<GpgKeyInfo>> GetGpgSecretKeysAsync();
```

Comment:
I had a look at the implementation and I have a feeling these names don't clearly communicate the purpose. `GetDefaultKey` appears to load the configured setting, whilst `GetGpgSecretKeysAsync` is actually loading and parsing keys.

Reply:
Yes GetDefaultKey is how it gets which key to use from git config so that the * can be placed in the listing along with the default being selected if there is one set.  GetSeceretKeys is the go get the list of keys from GPG.  

https://git-scm.com/docs/git-config#Documentation/git-config.txt-usersigningKey and https://git-scm.com/book/en/v2/Git-Tools-Signing-Your-Work 

I can rename to GetUserSigningKey if you want.  No I did not implement the ssh functionality in this part yet.

>user.signingKey
If [git-tag[1]](https://git-scm.com/docs/git-tag) or [git-commit[1]](https://git-scm.com/docs/git-commit) is not selecting the key you want it to automatically when creating a signed tag or commit, you can override the default selection with this variable. This option is passed unchanged to gpg’s --local-user parameter, so you may specify a key using any method that gpg supports. If gpg.format is set to ssh this can contain the path to either your private ssh key or the public key when ssh-agent is used. Alternatively it can contain a public key prefixed with key:: directly (e.g.: "key::ssh-rsa XXXXXX identifier"). The private key needs to be available via ssh-agent. If not set git will call gpg.ssh.defaultKeyCommand (e.g.: "ssh-add -L") and try to use the first key available. For backward compatibility, a raw key which begins with "ssh-", such as "ssh-rsa XXXXXX identifier", is treated as "key::ssh-rsa XXXXXX identifier", but this form is deprecated; use the key:: form instead.